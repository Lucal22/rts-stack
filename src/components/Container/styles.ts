import styled, { css, DefaultTheme } from 'styled-components';

export type ContainerStyleProps = {
  background?: 'white' | 'gray';
  height: 'full' | '100';
};

const backGround = {
  white: (theme: DefaultTheme) => css`
    background-color: ${theme.background.white};
    color: ${theme.colors.blackColor};
  `,
  gray: (theme: DefaultTheme) => css`
    background-color: ${theme.background.gray};
    color: ${theme.colors.whiteColor};
  `,
};

export const Container = styled.div<ContainerStyleProps>`
  ${({ theme, background, height }) => css`
    display: block;
    margin: 0 auto;
    padding: 0 1rem;
    min-height: ${height == 'full' ? '100vh' : '100%'};
    width: ${theme.screen.size.xSmall};
    @media (min-width: ${theme.screen.size.small}) {
      width: ${theme.screen.size.small};
    }
    @media (min-width: ${theme.screen.size.medium}) {
      width: ${theme.screen.size.medium};
    }
    @media (min-width: ${theme.screen.size.large}) {
      width: ${theme.screen.size.large};
    }
    ${background ? backGround[background](theme) : null}
  `}
`;
