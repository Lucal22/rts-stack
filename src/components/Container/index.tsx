import * as Styled from './styles';
import { ContainerStyleProps } from './styles';

export type ContainerProps = ContainerStyleProps & {
  children: React.ReactNode;
};

export default function Container({
  children,
  background,
  height,
}: ContainerProps) {
  return (
    <Styled.Container height={height} background={background}>
      {children}
    </Styled.Container>
  );
}
