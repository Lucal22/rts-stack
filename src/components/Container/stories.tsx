import { ReactNode } from 'react';
import Container from '.';
import { ContainerStyleProps } from './styles';

export default {
  title: 'Container',
  component: Container,
  args: {
    children: 'OlÃ¡ mundo',
    background: 'gray',
    height: 'full',
  },
  argTypes: {
    children: { type: 'string' },
  },
};

export const Template = (
  args: JSX.IntrinsicAttributes & ContainerStyleProps & { children: ReactNode },
) => <Container {...args} />;
