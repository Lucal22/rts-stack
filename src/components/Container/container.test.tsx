import { renderTheme } from '../../utils/renderTheme';
import Container from './index';

describe('<Container />', () => {
  it('should render with background white', () => {
    const { container } = renderTheme(
      <Container background={'white'} height={'full'}>
        <h1>Olá mundo</h1>
      </Container>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with background gray', () => {
    const { container } = renderTheme(
      <Container background={'gray'} height={'100'}>
        <h1>Olá mundo</h1>
      </Container>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with background gray', () => {
    const { container } = renderTheme(
      <Container height={'100'}>
        <h1>Olá mundo</h1>
      </Container>,
    );
    expect(container).toMatchSnapshot();
  });
});
