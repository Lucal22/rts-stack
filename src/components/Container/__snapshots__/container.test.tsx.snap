// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render with background gray 1`] = `
.c0 {
  display: block;
  margin: 0 auto;
  padding: 0 1rem;
  min-height: 100%;
  width: 360px;
  background-color: #6b7688;
  color: #ffffff;
}

@media (min-width:480px) {
  .c0 {
    width: 480px;
  }
}

@media (min-width:780px) {
  .c0 {
    width: 780px;
  }
}

@media (min-width:1280px) {
  .c0 {
    width: 1280px;
  }
}

<div>
  <div
    class="c0"
    height="100"
  >
    <h1>
      Olá mundo
    </h1>
  </div>
</div>
`;

exports[`<Container /> should render with background gray 2`] = `
.c0 {
  display: block;
  margin: 0 auto;
  padding: 0 1rem;
  min-height: 100%;
  width: 360px;
}

@media (min-width:480px) {
  .c0 {
    width: 480px;
  }
}

@media (min-width:780px) {
  .c0 {
    width: 780px;
  }
}

@media (min-width:1280px) {
  .c0 {
    width: 1280px;
  }
}

<div>
  <div
    class="c0"
    height="100"
  >
    <h1>
      Olá mundo
    </h1>
  </div>
</div>
`;

exports[`<Container /> should render with background white 1`] = `
.c0 {
  display: block;
  margin: 0 auto;
  padding: 0 1rem;
  min-height: 100vh;
  width: 360px;
  background-color: #ffffff;
  color: #030303;
}

@media (min-width:480px) {
  .c0 {
    width: 480px;
  }
}

@media (min-width:780px) {
  .c0 {
    width: 780px;
  }
}

@media (min-width:1280px) {
  .c0 {
    width: 1280px;
  }
}

<div>
  <div
    class="c0"
    height="full"
  >
    <h1>
      Olá mundo
    </h1>
  </div>
</div>
`;
